File: src/main/java/twao/villages/AllyVillage.java
package twao.villages;

import twao.Player;

public class AllyVillage extends Village {
    private final Player owner;

    /**
     * Initializes {@code owner}'s village at ({@code x}|{@code y}).
     * @param x
     * @param y
     * @param owner
     */
    public AllyVillage(int x, int y, Player owner) {
        super(x, y);
        this.owner = owner;
    }

    public Player getOwner() { return owner; }
}

File: src/main/kotlin/io/github/stiv3ns/twactionorganizer/core/villages/AllyVillage.kt
package io.github.stiv3ns.twactionorganizer.core.villages

import io.github.stiv3ns.twactionorganizer.core.Player

data class AllyVillage(
    override val x: Int,
    override val y: Int,
    val owner: Player
) : Village(x, y) {
    override fun toString() = "$x|$y"
}